File: sample-videochat-webrtc/sample-videochat-webrtc/Components/Settings/VideoSettingsViewController/BaseSettingsCell.m
//
//  BaseSettingsCell.m
//  sample-videochat-webrtc
//
//  Created by Injoit on 2/25/19.
//  Copyright © 2019 Quickblox. All rights reserved.
//

#import "BaseSettingsCell.h"

@implementation BaseSettingsCell

+ (NSString *)identifier {
    return NSStringFromClass([self class]);
}

+ (UINib *)nib {
    return [UINib nibWithNibName:[self identifier] bundle:nil];
}

- (void)setModel:(BaseItemModel *)model {
    
    _model = model;
    self.label.text = model.title;
}

@end

File: sample-videochat-webrtc-swift/sample-videochat-webrtc-swift/Components/Settings/BaseSettingsViewController/BaseSettingsCell.swift
//
//  BaseSettingsCell.swift
//  sample-videochat-webrtc-swift
//
//  Created by Injoit on 12/11/18.
//  Copyright © 2018 QuickBlox. All rights reserved.
//

import UIKit

protocol SettingsCellDelegate: class {
    func cell(_ cell: BaseSettingsCell, didChageModel model: BaseItemModel)
}

class BaseSettingsCell: UITableViewCell {
    //MARK: - IBOutlets
    @IBOutlet weak var label: UILabel!
    
    //MARK: - Properties
    var model = BaseItemModel() {
        didSet {
            updateModel(model)
        }
    }
    weak var delegate: SettingsCellDelegate?
    
    //MARK: - Class Methods
    class func identifier() -> String? {
        guard let cellIdentifier = NSStringFromClass(BaseSettingsCell.self).components(separatedBy: ".").last else {
            return nil
        }
        return cellIdentifier
    }
    
    class func nib() -> UINib? {
        return UINib(nibName: NSStringFromClass(BaseSettingsCell.self), bundle: nil)
    }
    
    //MARK: - Public Methods
    func updateModel(_ model: BaseItemModel) {
        label.text = model.title
    }
}

